src/main/groovy/org/spongepowered/gradle/ore/OreDeployTask.groovy

```diff
@@ -0,0 +1,156 @@
+        }
+
+        logger.quiet('Publishing ' + plugin.name + ' to ' + instanceUrl + '.')
+        logger.quiet('  Recommended: ' + recommended)
+        logger.quiet('  Channel: ' + channel)
+
+        if (instanceUrl.endsWith("/")) {
+            instanceUrl = instanceUrl.substring(0, instanceUrl.length() - 1)
+        }
+        URL projectUrl
+        try {
+            String pluginId = project.sponge.plugin.id
+            String str = instanceUrl + '/api/projects/' + pluginId + '/versions/' + project.version
+            projectUrl = new URL(str)
+        } catch (MalformedURLException e) {
+            throw new InvalidUserDataException("Invalid project URL", e)
+        }
+
+        if (apiKey == null) {
+            apiKey = project.property('oreDeploy.apiKey')
+        }
+        if (channel == null) {
+            throw new InvalidUserDataException("Missing channel name.")
+        }
+        HttpEntity requestEntity = MultipartEntityBuilder.create()
+            .addPart('apiKey', new StringBody(apiKey))
+            .addPart('channel', new StringBody(channel))
+            .addPart('recommended', new StringBody(recommended as String))
+            .addPart('pluginFile', new FileBody(plugin.file))
+            .addPart('pluginSig', new FileBody(sig.file))
+            .build()
+
+        HttpURLConnection conn = null
+        InputStream is = null
+        try {
+            conn = (HttpURLConnection) projectUrl.openConnection()
+            conn.doOutput = true
+            conn.requestMethod = 'POST'
+            conn.useCaches = false
+            conn.readTimeout = 10000
+            conn.connectTimeout = 10000
+            conn.setRequestProperty('Connection', 'Keep-Alive')
+            conn.setRequestProperty('Cache-Control', 'no-cache')
+            conn.setRequestProperty('Content-Length', requestEntity.contentLength as String)
+            conn.setRequestProperty('Content-Type', requestEntity.contentType.value)
+
+            conn.outputStream.withStream { requestEntity.writeTo(it) }
+
+            conn.connect()
+            if (conn.responseCode != HttpURLConnection.HTTP_CREATED) {
+                logger.error('[failure] ' + conn.responseCode + ' ' + conn.responseMessage)
+                is = conn.errorStream
+                printErrorStream is
+                is.close()
+                is = null
+                throw new GradleException()
+            } else {
+                is = conn.inputStream
+                def json = new JsonSlurper().parse(is, 'UTF-8')
+                logger.quiet('[success] ' + instanceUrl + (json.href as String))
+                is.close()
+                is = null
+            }
+        } catch (InterruptedIOException | SocketException ignored) {
+            throw new GradleException('Failed to connect to Ore.')
+        } catch (SSLHandshakeException ignored) {
+            logger.error('[failure] Outdated Java!')
```

Comment:
I think it would make more sense if we include this message in the exception below